universal-application-tool-0.0.1/app/assets/javascripts/admin_validation.ts

```diff
@@ -0,0 +1,65 @@
+/** The validation controller provides basic client-side validation of admin form fields. */
+class AdminValidationController {
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_CREATE = '#question-settings input[name="newOptions[]"]';
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_EDIT = '#question-settings input[name="options[]"]';
+  static readonly MULTI_OPTION_QUESTION_OPTION_CLASS = ".cf-multi-option-input";
+
+  constructor() {
+    // Attach listener to admin program edit form.
+    const adminProgramEditForm = document.getElementById("question-form");
+    if (adminProgramEditForm) {
+      adminProgramEditForm.addEventListener("submit", (event) => this.attemptSubmit(event));
+    }
+  }
+
+  private attemptSubmit(event: Event) {
+    this.validateForm();
+    if (!this.validateForm()) {
+      event.preventDefault();
+    }
+  }
+
+  private validateForm(): boolean {
+    return this.validateMultiOptionQuestionCreate() && this.validateMultiOptionQuestionEdit();
+  }
+
+  private updateFieldErrorState(element: Element, fieldName: string, isValid: boolean) {
+    const errorDiv = element.parentElement!.querySelector(fieldName + '-error');
+    if (errorDiv) {
+      errorDiv.classList.toggle('hidden', isValid);
+    }
+
+    // Also toggle the border on error inputs (if applicable).
+    const field = element.parentElement!.querySelector('input');
+    if (field) {
+      field.classList.toggle('border-red-600', !isValid);
+    }
+  }
+
+  /** Validates that there are no empty options. Returns true if all fields are valid.  */
+  private validateMultiOptionQuestionOptions(options: HTMLInputElement[]) {
+    let multiOptionIsValid = true;
+    for (const option of options) {
+      const inputIsValid = option.value !== '';
+      this.updateFieldErrorState(option, AdminValidationController.MULTI_OPTION_QUESTION_OPTION_CLASS, inputIsValid);

```

Comment:
suggestion (blocking): the only usage of `MULTI_OPTION_QUESTION_OPTION_CLASS` concatenates it with `"-error"`, recommend adding that bit to the constant and removing the concatenation

Reply:
Fixed.